src/attributes.jl

```diff
@@ -449,6 +449,39 @@ struct RawParameter <: AbstractOptimizerAttribute
     name::Any
 end
 
+### Callbacks
+
+"""
+    struct OptimizeInProgress{AttrType<:AnyAttribute} <: Exception
+        attr::AttrType
+    end
+Error thrown from optimizer when `MOI.get(optimizer, attr)` is called inside an
+[`AbstractCallback`](@ref) while it is only defined once [`optimize!`](@ref) has
+completed. This is only defined if `is_set_by_optimize(attr)` is `true`.
+"""
+struct OptimizeInProgress{AttrType<:AnyAttribute} <: Exception
+    attr::AttrType
+end
+
+function Base.showerror(io::IO, err::OptimizeInProgress)
+    print(io, typeof(err), ": Cannot get result as the `MOI.optimize!` has not",
+          " finished.")
+end
+
+"""
+    abstract type AbstractCallback <: AbstractOptimizerAttribute end
```

Comment:
Wouldn't `AbstractCallback` be a `Submittable` instead? That is, one can usually more than one callback to a solver?

Reply:
That could work. That would allow callbacks to be added independently from different places that may not know about each other (as we discussed with @juan-pablo-vielma ).
One drawback is that since `submit` does not return any index, we would not be allowed to remove a callback.
With the attribute, setting it to another callback removes the previous one.

Reply:
Thanks, I have now learned that most solvers only work with a single callback (different from SCIP), so this choice makes sense for the solver-independent infrastructure.